# Generated by Django 4.0.1 on 2022-02-09 19:26

from django.db import migrations, models
import django.db.models.deletion
import systems.models.fields


class Migration(migrations.Migration):

    initial = True

    dependencies = [
        ('dataset', '0003_alter_dataset_groups'),
        ('group', '0003_alter_group_parent'),
    ]

    operations = [
        migrations.CreateModel(
            name='RemoteAIModel',
            fields=[
                ('created', models.DateTimeField(editable=False, null=True)),
                ('updated', models.DateTimeField(editable=False, null=True)),
                ('name', models.CharField(max_length=100, primary_key=True, serialize=False)),
                ('config', systems.models.fields.EncryptedDataField(default={}, editable=False)),
                ('provider_type', models.CharField(editable=False, max_length=128, null=True)),
                ('variables', systems.models.fields.EncryptedDataField(default={}, editable=False)),
                ('training_percentage', models.FloatField(default=0.7)),
                ('precision', models.FloatField(null=True)),
                ('recall', models.FloatField(null=True)),
                ('f1_score', models.FloatField(null=True)),
                ('dataset', models.ForeignKey(editable=False, null=True, on_delete=django.db.models.deletion.SET_NULL, related_name='%(class)s_relation', to='dataset.dataset')),
                ('groups', models.ManyToManyField(related_name='%(class)s_relations', to='group.Group')),
            ],
            options={
                'verbose_name': 'remote ai model',
                'verbose_name_plural': 'remote ai models',
                'db_table': 'remote_ai_remote_ai_model',
                'ordering': ['name', '-created'],
                'abstract': False,
            },
        ),
    ]
